File: php/folderprogram/Smartphone.php
<?php
require_once "Gadget.php";

class Smartphone extends Gadget {
    private $sistemOperasi;
    private $memori;
    private $kamera;

    public function __construct($kode = "", $nama = "", $harga = 0, $deskripsi = "",
                                $jenis = "", $garansi = "", $status = "", $stok = 0,
                                $sistemOperasi = "", $memori = "", $kamera = "", $foto = "") {
        // Panggil konstruktor parent dan teruskan semua parameter, termasuk $foto
        parent::__construct($kode, $nama, $harga, $deskripsi, $jenis, $garansi, $status, $stok, $foto);
        $this->sistemOperasi = $sistemOperasi;
        $this->memori = $memori;
        $this->kamera = $kamera;
    }

    // Getter & Setter
    public function getSistemOperasi() { return $this->sistemOperasi; }
    public function setSistemOperasi($sistemOperasi) { $this->sistemOperasi = $sistemOperasi; }

    public function getMemori() { return $this->memori; }
    public function setMemori($memori) { $this->memori = $memori; }

    public function getKamera() { return $this->kamera; }
    public function setKamera($kamera) { $this->kamera = $kamera; }
    
    // getFoto() dan setFoto() tidak perlu lagi karena sudah ada di kelas Gadget
    
    public function displayInfo() {
        parent::displayInfo();
        echo "Sistem Operasi: {$this->sistemOperasi}<br>";
        echo "Memori: {$this->memori}<br>";
        echo "Kamera: {$this->kamera}<br>";
    }
}
?>

File: python/folderprogram/Smartphone.py
# Smartphone.py

from Gadget import Gadget

class Smartphone(Gadget):
    def __init__(self, kode, nama, harga, deskripsi, jenis, garansi, status, stok,
                 sistemOperasi, memori, kamera):
        # FIXED: Meneruskan semua argumen yang diperlukan ke parent class
        super().__init__(kode, nama, harga, deskripsi, jenis, garansi, status, stok)
        self.sistemOperasi = sistemOperasi
        self.memori = memori
        self.kamera = kamera

    def display_info(self):
        super().display_info()
        print(f"Sistem Operasi: {self.sistemOperasi}")
        print(f"Memori: {self.memori}")
        print(f"Kamera: {self.kamera}")
        print("-" * 30) # Pemisah antar data

    # FIXED: Getter dan Setter dipindahkan ke dalam kelas
    def get_sistemOperasi(self):
        return self.sistemOperasi

    def set_sistemOperasi(self, sistemOperasi):
        self.sistemOperasi = sistemOperasi

    def get_memori(self):
        return self.memori

    def set_memori(self, memori):
        self.memori = memori

    def get_kamera(self):
        return self.kamera

    def set_kamera(self, kamera):
        self.kamera = kamera